tests/zstreamtest.c

```diff
@@ -2066,6 +2066,90 @@ static int basicUnitTests(U32 seed, double compressibility, int bigTests)
     }
     DISPLAYLEVEL(3, "OK \n");
 
+    /* Test Sequence Validation */
+    DISPLAYLEVEL(3, "test%3i : Testing sequence validation: ", testNb++);
+    {
+        ZSTD_CCtx* cctx = ZSTD_createCCtx();
+
+        /* Test minMatch >= 4, matchLength < 4 */
+        {
+            size_t srcSize = 11;
+            void* const src = CNBuffer;
+            size_t dstSize = ZSTD_compressBound(srcSize);
+            void* const dst = compressedBuffer;
+            size_t const kNbSequences = 4;
+            ZSTD_Sequence* sequences = malloc(sizeof(ZSTD_Sequence) * kNbSequences);
+
+            memset(src, 'x', srcSize);
+
+            sequences[0] = (ZSTD_Sequence) {1, 1, 3, 0};
+            sequences[1] = (ZSTD_Sequence) {1, 0, 3, 0};
+            sequences[2] = (ZSTD_Sequence) {1, 0, 3, 0};
+            sequences[3] = (ZSTD_Sequence) {0, 1, 0, 0};
+
+            /* Test with sequence validation */
+            CHECK_Z(ZSTD_CCtx_setParameter(cctx, ZSTD_c_minMatch, 5));
+            CHECK_Z(ZSTD_CCtx_setParameter(cctx, ZSTD_c_blockDelimiters, ZSTD_sf_explicitBlockDelimiters));
+            CHECK_Z(ZSTD_CCtx_setParameter(cctx, ZSTD_c_validateSequences, 1));
+
+            cSize = ZSTD_compressSequences(cctx, dst, dstSize,
+                                   sequences, kNbSequences,
+                                   src, srcSize);
+
+            CHECK(!ZSTD_isError(cSize), "Should throw an error"); /* maxNbSeq is too small and an assert will fail */
```

Comment:
Can you add a second check to ensure it's the expected error code? Example: https://github.com/facebook/zstd/blob/64963dcbd6162c52ba9273bb55d78c7a442b12f4/tests/zstreamtest.c#L1895-L1898

Reply:
Added! Note: validation is by default turned on in debug mode so I could not add this second check for the tests where validation was turned off (since a different error is produced depending on the debug level).

Reply:
Note: 
`debug` mode typically means `DEBUGLEVEL=1`, which enables `assert()`.
As a consequence, the program may `abort()` on some detected conditions, while it would not in `release` mode.

I don't think we ever had (so far) a difference in error code depending on `debug` vs `release`.
This would be a first, and indeed makes tests more complex to set up, since the outcome depends on `debug` vs `release` status.

I believe it's better to avoid this complication, and revert to the simpler idea that `debug` can `abort()`, but does not change produced error codes.

Reply:
@Cyan4973 @daniellerozenblit We just need to align these two error codes:
* https://github.com/facebook/zstd/blob/a0e3f2e4c0ff214323f7c9bab03d61a6a3124149/lib/compress/zstd_compress.c#L6342
* https://github.com/facebook/zstd/blob/a0e3f2e4c0ff214323f7c9bab03d61a6a3124149/lib/compress/zstd_compress.c#L6172

Perhaps we can add a new error code, something like "invalid external sequences"? Or move both of them to `corruption_detected`?

Reply:
I agree @embg that it would be best to align these error codes, since they are signaling the same issue.

Any preference @Cyan4973 between creating a new error code or using `corruption_detected` for both?

Reply:
I think it's valid to create a new error code for a more accurate description of the issue.